### ISSUE TITLE
Protected Environments not supported

### ISSUE THREAD

**Issue by nastrofaction** [python-gitlab/python-gitlab#1130]
## Description of the problem, including code/CLI snippet
Per API: https://docs.gitlab.com/ee/api/protected_environments.html we should be able to create protected environment but the code fails as it isn't supported

## Expected Behavior
```
project.protectedenvironments.create({
                'name': env,
                'deploy_access_levels': [{"access_level": gitlab.MAINTAINER_ACCESS}]
            })
```
should create the protected environment

## Actual Behavior
Doesn't work
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/gitlab/base.py", line 59, in __getattr__
    return self.__dict__["_updated_attrs"][name]
KeyError: 'protectedenvironments'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/gitlab/base.py", line 62, in __getattr__
    value = self.__dict__["_attrs"][name]
KeyError: 'protectedenvironments'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/gitlab/base.py", line 82, in __getattr__
    return self.__dict__["_parent_attrs"][name]
KeyError: 'protectedenvironments'
```

## Specifications

  - python-gitlab version: python-gitlab-2.3.1
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com): gitlab.com

**Comment by max-wittig** [comment]
I don't have access to Gitlab Premium. That's why I'm not really implementing these features, but I would be open to review any MRs regarding EE features.

**Comment by nastrofaction** [comment]
Thanks for responding @max-wittig For now I am using cURL to complete my automation through API for protected environments. Once this is complete; I'll try to come back and submit MR for this.

**Comment by filip-malczak-bny** [comment]
@nastrofaction is there any update on this?

**Comment by calve** [comment]
I added support for Protected Environments in https://github.com/python-gitlab/python-gitlab/pull/2084 which is merged but not released yet

Do note that the API use the name `protected_environments` and not `protectedenvironments` as specified in this issue

**Comment by nickbroon** [comment]
Protected environment support is available since [v3.6.0](https://github.com/python-gitlab/python-gitlab/releases/tag/v3.6.0). Should this issues now be closed?

**Comment by nejch** [comment]
Oh, completely missed that one, good to do some cleanup here. Thanks a lot @nickbroon!
